File: Data Structures & Algorithms/string-matching-in-an-array/submission-1.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        words.sort(key=len)
        res = []

        for i, w in enumerate(words):
            for j in range(i + 1, len(words)):
                if w in words[j]:
                    res.append(w)
                    break
        
        return res

File: Data Structures & Algorithms/longest-common-prefix/submission-1.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        Array.Sort(strs);
        // compare first and last
        var first = strs[0];
        var last = strs[strs.Length - 1];
        string res = "";

        for (int i = 0; i < Math.Min(first.Length, last.Length); i++) {
            if (first[i] != last[i]) {
                break;
            }

            res += first[i];
        }

        return res;
    }
}